File: server/apps/nexotype/schemas/knowledge_graph/evidence_assertion_schemas.py
from pydantic import BaseModel, ConfigDict
from datetime import datetime

# ==========================================
# Request Schemas
# ==========================================

class EvidenceAssertionCreate(BaseModel):
    """Schema for creating a new evidence assertion"""
    relationship_table: str
    relationship_id: int
    source_id: int
    confidence_score: float


class EvidenceAssertionUpdate(BaseModel):
    """Schema for updating an evidence assertion"""
    relationship_table: str = None
    relationship_id: int = None
    source_id: int = None
    confidence_score: float = None


# ==========================================
# Response Schemas
# ==========================================

class EvidenceAssertionDetail(BaseModel):
    """Schema for evidence assertion details"""
    id: int
    relationship_table: str
    relationship_id: int
    source_id: int
    confidence_score: float
    created_at: datetime
    updated_at: datetime | None = None

    model_config = ConfigDict(from_attributes=True)


class EvidenceAssertionListResponse(BaseModel):
    """Response schema for listing evidence assertions"""
    success: bool
    data: list[EvidenceAssertionDetail] | None = None
    count: int = 0
    error: str | None = None


class EvidenceAssertionResponse(BaseModel):
    """Response schema for single evidence assertion operations"""
    success: bool
    data: EvidenceAssertionDetail | None = None
    error: str | None = None


class MessageResponse(BaseModel):
    """Simple message response"""
    success: bool
    message: str | None = None
    error: str | None = None

File: client/src/modules/nexotype/schemas/knowledge_graph/evidence-assertion.schemas.ts
import { z } from 'zod';

// ==========================================
// Full Schema (GET responses)
// ==========================================

export const EvidenceAssertionSchema = z.object({
  id: z.number(),
  relationship_table: z.string().min(1).max(50),
  relationship_id: z.number().int(),
  source_id: z.number().int(),
  confidence_score: z.number(),
  created_at: z.string(),
  updated_at: z.string().nullable(),
});

// ==========================================
// Input Schemas (POST/PUT)
// ==========================================

export const CreateEvidenceAssertionSchema = z.object({
  relationship_table: z.string().min(1).max(50),
  relationship_id: z.number().int(),
  source_id: z.number().int(),
  confidence_score: z.number(),
});

export const UpdateEvidenceAssertionSchema = z.object({
  relationship_table: z.string().min(1).max(50).optional(),
  relationship_id: z.number().int().optional(),
  source_id: z.number().int().optional(),
  confidence_score: z.number().optional(),
});

// ==========================================
// Types
// ==========================================

export type EvidenceAssertion = z.infer<typeof EvidenceAssertionSchema>;
export type CreateEvidenceAssertion = z.infer<typeof CreateEvidenceAssertionSchema>;
export type UpdateEvidenceAssertion = z.infer<typeof UpdateEvidenceAssertionSchema>;

// ==========================================
// Response Types
// ==========================================

export type EvidenceAssertionResponse = {
  success: boolean;
  data?: EvidenceAssertion;
  error?: string;
};

export type EvidenceAssertionsResponse = {
  success: boolean;
  data?: EvidenceAssertion[];
  count?: number;
  error?: string;
};
